### PR TITLE
Language Dropdown Fixes

### PR DESCRIPTION
**Pull request by neilsutcliffe**
The language dropdown did not display the selected value. Also small fixes to the language. Might cause a shake up, but I have tested locally with the docusaurus Repo. Could not test npm run build due to my OS.

### CODE REVIEW

File: lib/core/nav/HeaderNav.js [hunk 1]
```diff
@@ -126,31 +134,24 @@ class HeaderNav extends React.Component {
...
-            ") is enabled when maybe they should not be. \nThus my internal id for this doc is: '" +
-            id +
-            "'. Please file an issue for this possible bug on GitHub.";
+        if (id != link.doc) {
+          throw new Error(
+            "It looks like you've enabled language support, but haven't provided translated files. The document with id: '" +
+              id +
+              "' doesn't exist."
+          );
         }
```
Comment (neilsutcliffe) [inline lib/core/nav/HeaderNav.js]: This must be a remnant from another pull request. Is it a bad change?

File: website/core/Footer.js [hunk 2]
```diff
@@ -21,10 +21,6 @@ const GithubButton = props => (
   </a>
 );
 
-GithubButton.propTypes = {
-  config: React.PropTypes.object
-};
-
```
Comment (neilsutcliffe) [inline website/core/Footer.js]: Only place we use it. It was logging an error each build. It isn't worth importing the 'proper' way unless we plan to use proptypes in more places than this IMHO.

File: lib/core/nav/HeaderNav.js (hunk 1 above)
Comment (JoelMarcey) [reply lib/core/nav/HeaderNav.js]: I added this in a previous commit.

**Comment by JoelMarcey** [comment]
@neilsutcliffe Hi! Can you rebase this as well? 

And then I want to pull it in locally and test on a couple of sites.

**Comment by yangshun** [comment]
@neilsutcliffe Are you available to rebase this PR and fix the conflicts? If not, it's all fine and we can take over!

**Comment by neilsutcliffe** [comment]
It's probably best if you close this and create it as a new PR TBH. It will have some unintended changes the way I did it.

Deffo refer to it if you need to though.
